From a9fb1f78ad72fdc070f5c61b003e4a13d1c46c68 Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
Date: Sat, 17 Jan 2015 19:55:19 -0800
Subject: [PATCH 2002/2002] On darwin, build ppc slices of the compiler runtime
 if requested and supported by the SDK and toolchain

Signed-off-by: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
(cherry picked from commit 1fe76a2b12f4837c49237cdca6803e5f43e00f7c)

Conflicts:
	make/platform/clang_darwin.mk
---
 make/platform/clang_darwin.mk | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git llvm_release_34/projects/compiler-rt/make/platform/clang_darwin.mk macports_release_34/projects/compiler-rt/make/platform/clang_darwin.mk
index c13063c..5adebfe 100644
--- llvm_release_34/projects/compiler-rt/make/platform/clang_darwin.mk
+++ macports_release_34/projects/compiler-rt/make/platform/clang_darwin.mk
@@ -53,11 +53,11 @@ UniversalArchs :=
 Configs += eprintf
 UniversalArchs.eprintf := $(call CheckArches,i386,eprintf)
 
-# Configuration for targetting 10.4. We need a few functions missing from
-# libgcc_s.10.4.dylib. We only build x86 slices since clang doesn't really
-# support targetting PowerPC.
+# Configuration for targeting 10.4. We need a few functions missing from
+# libgcc_s.10.4.dylib. We don't build ppc64 slices since clang doesn't really
+# support it..
 Configs += 10.4
-UniversalArchs.10.4 := $(call CheckArches,i386 x86_64,10.4)
+UniversalArchs.10.4 := $(call CheckArches,ppc i386 x86_64,10.4)
 
 # Configuration for targetting iOS for a couple of functions that didn't
 # make it into libSystem.
@@ -67,7 +67,7 @@ UniversalArchs.ios := $(call CheckArches,i386 x86_64 armv7,ios)
 # Configuration for targetting OSX. These functions may not be in libSystem
 # so we should provide our own.
 Configs += osx
-UniversalArchs.osx := $(call CheckArches,i386 x86_64,osx)
+UniversalArchs.osx := $(call CheckArches,ppc i386 x86_64,osx)
 
 # Configuration for use with kernel/kexts.
 Configs += cc_kext
@@ -80,7 +80,7 @@ UniversalArchs.cc_kext_ios5 := $(call CheckArches,x86_64 armv7,cc_kext_ios5)
 
 # Configurations which define the profiling support functions.
 Configs += profile_osx
-UniversalArchs.profile_osx := $(call CheckArches,i386 x86_64,profile_osx)
+UniversalArchs.profile_osx := $(call CheckArches,ppc i386 x86_64,profile_osx)
 Configs += profile_ios
 UniversalArchs.profile_ios := $(call CheckArches,i386 x86_64 armv7,profile_ios)
 
-- 
2.2.2

